references/recaptcha-enterprise/pipeline.sh

```diff
@@ -0,0 +1,324 @@
+                        {
+                            "name": "SITE_KEY",
+                            "value": "$SITEKEY_ALWAYS_1"
+                        }
+                    ],
+                    "apiProducts": [
+                        "RecaptchaEnterprise"
+                    ],
+                    "callbackUrl": "",
+                    "scopes": []
+                },
+                {
+                    "name": "app-recaptcha-enterprise-always0",
+                    "attributes": [
+                        {
+                            "name": "GCP_PROJECT_ID",
+                            "value": "$GCP_PROJECT_ID"
+                        },
+                        {
+                            "name": "SITE_KEY",
+                            "value": "$SITEKEY_ALWAYS_0"
+                        }
+                    ],
+                    "apiProducts": [
+                        "RecaptchaEnterprise"
+                    ],
+                    "callbackUrl": "",
+                    "scopes": []
+                }
+            ]
+        }
+    }
+}
+EOF
+}
+
+#################################
+### function: set_idp_env_var ###
+#################################
+set_functional_test_env_var() {
+
+    # use timestamp (parameter) to create a unique value
+    TEST_APP_CONSUMER_KEY="xkey-$1"
+    export TEST_APP_CONSUMER_KEY
+    TEST_APP_CONSUMER_SECRET='xsecret'
+    export TEST_APP_CONSUMER_SECRET
+}
+
+######################################
+### function: set_recaptcha_config ###
+######################################
+set_recaptcha_config() {
+
+    # Copy AM policy template
+    cp "$SCRIPTPATH"/templates/AM-SetReCaptchaMock.template.xml "$SCRIPTPATH"/sf-recaptcha-enterprise-v1/sharedflowbundle/policies/AM-SetReCaptchaMock.xml
+
+    # replace reCAPTCHA tag in AM policy
+    sed -i.bak "s|@IsReCaptchaMockEnabled@|$1|" "$SCRIPTPATH"/sf-recaptcha-enterprise-v1/sharedflowbundle/policies/AM-SetReCaptchaMock.xml
+
+    # remove .bak files
+    rm "$SCRIPTPATH"/sf-recaptcha-enterprise-v1/sharedflowbundle/policies/AM-SetReCaptchaMock.xml.bak
+}
+
+### script execution starts here...
+
+# Check for required variables
+if [ -z "$GCP_PROJECT" ]; then
+  echo "The required env variable GCP_PROJECT is missing";
+  exit 1
+fi
+
+# Manage optional variables
+GCP_REGION=${GCP_REGION:-europe-west1}
+timestamp=$(date '+%s')
+export PATH="$PATH:$SCRIPTPATH/../../tools/apigee-sackmesser/bin"
+
+# default value for reCAPTCHA mock mode (default is 'true')
+DEFAULT_IS_RECAPTCHA_MOCK_ENABLED=true
+
+# is reCAPTCHA mock enabled (=true) or not (=false)
+IS_RECAPTCHA_MOCK_ENABLED="${IS_RECAPTCHA_MOCK_ENABLED:-"$DEFAULT_IS_RECAPTCHA_MOCK_ENABLED"}"
+
+# set reCAPTCHA config
+set_recaptcha_config "$IS_RECAPTCHA_MOCK_ENABLED"
+
+echo "[INFO] Deploying reCAPTCHA enterprise reference to Google API (For X/hybrid)"
+APIGEE_TOKEN=$(gcloud auth print-access-token);
+
+## Generate Service Account for Apigee shared flow
```

Comment:
please make it conditional as in the other SA dependent pipelines. The nightly does not have permissions to create SAs.

https://github.com/apigee/devrel/blob/3a44851e166762397094faac4fb3240707ab990b/references/cloud-run/pipeline.sh#L32

Reply:
I don't think we need the SA with permissions if the mock is enabled do we?

Reply:
If the mock is enabled we still need a service account but no need to set the right permission on it

Reply:
OK, done